docs/crash_reporting.md

```diff
@@ -0,0 +1,202 @@
+set -e
+export SOURCEMAP_FILE=./build/main.jsbundle.map # <- add this line to output sourcemaps
+# leave the rest of the script unchanged
+```
+
+Moving forward, you can find the source maps for your bundle on every iOS build.
+
+To find the path to your bundle file from XCode, display the Report Navigator on XCode and filter by `BUNDLE_FILE` for its location.
+
+The usual location is `~/Library/Developer/Xcode/DerivedData/YourAppName-verylonghash/Build/Intermediates.noindex/ArchiveIntermediates/YourAppName/BuildProductsPath/Release-iphoneos/main.jsbundle`, where `YourAppName` is the name of your app, and `verylonghash` is a 28 letter hash.
+
+To upload the source maps, run this from your React Native project:
+
+```bash
+export DATADOG_API_KEY= # fill with your API key
+export SERVICE=com.myapp # replace by your service name
+export VERSION=1.0.0 # replace by the version of your app in XCode
+export BUILD=100 # replace by the build of your app in XCode
+export BUNDLE_PATH= # fill with your bundle path
+
+yarn datadog-ci react-native upload --platform ios --service $SERVICE --bundle $BUNDLE_PATH --sourcemap ./build/main.jsbundle.map --release-version $VERSION --build-version $BUILD
+```
+
+#### Manually on each build (with Hermes)
+
+There is currently a bug in React Native that generates an incorrect source map when using Hermes.
+
+To resolve this, you need to add more lines **at the very end** of the build phase to generate a correct source map file.
+
+Edit your build phase like so:
+
+```bash
+set -e
+export SOURCEMAP_FILE=./build/main.jsbundle.map # <- add this line to output sourcemaps
+
+# keep the rest of the script unchanged
+
+# add these lines to compose the packager and compiler sourcemaps into one file
+REACT_NATIVE_DIR=../node_modules/react-native
+source "$REACT_NATIVE_DIR/scripts/find-node.sh"
+source "$REACT_NATIVE_DIR/scripts/node-binary.sh"
+"$NODE_BINARY" "$REACT_NATIVE_DIR/scripts/compose-source-maps.js" "$CONFIGURATION_BUILD_DIR/main.jsbundle.map" "$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH/main.jsbundle.map" -o "../$SOURCEMAP_FILE"
+```
+
+To upload the source map, run this from your React Native project root:
+
+```bash
+export DATADOG_API_KEY= # fill with your API key
+export SERVICE=com.myapp # replace by your service name
+export VERSION=1.0.0 # replace by the version of your app in XCode
+export BUILD=100 # replace by the build of your app in XCode
+export BUNDLE_PATH= # fill with your bundle path
+
+yarn datadog-ci react-native upload --platform ios --service $SERVICE --bundle $BUNDLE_PATH --sourcemap ./build/main.jsbundle.map --release-version $VERSION --build-version $BUILD
+```
+
+
+### Upload JavaScript source maps on Android builds
+
+#### Manually on each build
+
+On Android, the bundle file is located at `android/app/build/generated/assets/react/release/index.android.bundle` and the source map file is located at `android/app/build/generated/sourcemaps/react/release/index.android.bundle.map`.
```

Comment:
maybe we should also mention, that it will be not necessarily `release` folder in case if app has more comprehensive variants.

Reply:
```suggestion
On Android, the bundle file is located at `android/app/build/generated/assets/react/release/index.android.bundle` and the source map file is located at `android/app/build/generated/sourcemaps/react/release/index.android.bundle.map`. If your application has more comprehensive variants, there may not be a `release` folder.
```